### ISSUE TITLE
[WebGL2] SpatialMaterial not rendering properly

### ISSUE THREAD

**Issue by DominicSina** [godotengine/godot#17047]
**Godot version:**
Godot 3.0.1-rc1 and Godot 3.0 stable

**OS/device including version:**
OS: Windows 10 Home (64-bit)
Browser: Firefox Quantum 58.0.2 (64-bit)
Processor: Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz (4 CPUs), ~2.2GHz
GPU: Intel(R) HD Graphics 5500
Driver Name: igdumdim64.dll,igd10iumd64.dll,igd10iumd64.dll,igd12umd64.dll
Driver Version: 10.18.15.4248
OpenGL ES 3.0 Renderer: Mozilla

**Issue description:**
What I've done:
I have imported a .glb scene(made in Blender with [Khronos](https://github.com/KhronosGroup/glTF-Blender-Exporter) exporter) into Godot, cleared the inheritance(I don't think glb is causing issues here, just thought I should mention it) and added scripts to initialize the SpatialMaterial of both MeshInstances. Each mesh also has a script that will change the albedo of its material every five seconds. I noticed two problems in the HTML5 export that don't happen if I run the project in the editor. 

Problem1:
If I run the project in the editor everything seems to work and each mesh changes its color individually. In the HTML5 export all meshes seem to share a color. It seems to me that this is not what was intended, the meshes should not be affected by the albedo of the others. 

Problem2:
If I turn on transparency (either `set_feature(FEATURE_TRANSPARENT,true)` or `params_blend_mode=m.BLEND_MODE_ADD`) then it seems that all meshes share the color of the mesh that is the closest(furthest?) to the camera. This leads to a strange effect where movement of the camera can affect the coloration of a mesh. I think this problem is related to the first one, since the meshes still share their color. In my scene both cubes swap between being rendered with the material of the first and the second cube. Again this seems to be the same underlying issue as problem 1 to me, it simply produces other side effects when combined with the particular rendering order that transparency requires.

**Steps to reproduce:**
For problem 1 simply run the reproduction project once as HTML5 export and once in the editor, since the material albedos change every 5s the difference in behavior will become apparent almost immediately. 

For problem 2 in InitMaterial.gd change how the materials are initialized. Uncomment either 
`m2.set_feature(m2.FEATURE_TRANSPARENT,true)` AND `m.set_feature(m2.FEATURE_TRANSPARENT,true)` or
`m2.params_blend_mode=m2.BLEND_MODE_ADD` AND `m.params_blend_mode=m.BLEND_MODE_ADD`. Afterwards run the project and rotate(click and drag) the camera around the two cubes. When rotating in the HTML5 export it becomes clear that the colors change much faster than every 5s, especially if you quickly move back and forth around the threshold where both cubes are almost the same distance. If needed change the timer in ColorFlickeringCube.gd to be a bit slower.

**Minimal reproduction project:**
[ColorFlickering.zip](https://github.com/godotengine/godot/files/1759257/ColorFlickering.zip)

**Comment by DominicSina** [comment]
I am starting to think this has something to do with the firefox version that is being used. I remember it working correctly in the past. Going back to previous Godot versions didn't solve my problem though. So I've tested a couple different browsers on the same system described above.

Browsers with the bug:
Firefox Quantum 58.0.2 (64-bit) (OpenGL ES 3.0 Renderer: Mozilla)
Firefox Quantum 59.0.2 (64-bit) (OpenGL ES 3.0 Renderer: Mozilla)

Browsers without the bug:
Chrome Version 65.0.3325.181 (Official Build) (64-bit) (OpenGL ES 3.0 Renderer: WebKit WebGL)


I've also tested it on a Linux system that I have in a virtual machine and the bug doesn't appear, even though it's the same browser version:
OS: Ubuntu 16.04 LTS (64 bit)
Graphics: llvmpipe (LLVM 5.0, 256 bits)
CPU: Intel® Core™ i5-5200U CPU @ 2.20GHz
Browser: Firefox Quantum 59.0.2 (64-bit) (OpenGL ES 3.0 Renderer: Mozilla)

Was someone able to reproduce this bug with the sample project?
